program/src/processor/deactivate_stake.rs

```diff
@@ -107,14 +108,10 @@ pub fn process_deactivate_stake(
 
     // Validate the amount.
 
-    require!(amount > 0, StakeError::InvalidAmount);
-
     require!(stake.amount >= amount, StakeError::InsufficientStakeAmount);
 
-    let max_deactivation_amount = get_max_deactivation_amount(
-        config.token_amount_delegated,
-        config.max_deactivation_basis_points,
-    )?;
+    let max_deactivation_amount =
+        get_max_deactivation_amount(stake.amount, config.max_deactivation_basis_points)?;
```

Comment:
oops, I didn't even realize this was incorrect before, nice fix!

Reply:
Yeah, it was something that we picked up with Joe went we went through the docs again last week. 😀